tests/components/habitica/test_todo.py

```diff
@@ -276,21 +256,19 @@ async def test_update_todo_item(
         blocking=True,
     )
 
-    mock_call = mock_called_with(
-        mock_habitica, "PUT", f"{DEFAULT_URL}/api/v3/tasks/{uid}"
+    task = Task(
+        notes="test-description",
+        text="test-summary",
     )
-    assert mock_call
-    assert json.loads(mock_call[2]) == {
-        "date": date,
-        "notes": "test-description",
-        "text": "test-summary",
-    }
+    if date:
+        task["date"] = date
```

Comment:
We don't want if statements in the tests as they make unclear what we are testing